### PR TITLE
Add changes from rector rule RemoveDeadStmtRector

### PR DESCRIPTION
**Pull request by Harfusha**
https://github.com/cakephp/cakephp/issues/17756

### CODE REVIEW

File: tests/TestCase/Database/Query/UpdateQueryTest.php [hunk 1]
```diff
@@ -471,13 +471,9 @@ public function testJsonValue(): void
         $driver = $this->connection->getDriver();
         $version = $this->connection->getDriver()->version();
         if ($driver instanceof Postgres && version_compare($version, '12.0.0', '<')) {
-            $skip == true;
```
Comment (dereuromark) [inline tests/TestCase/Database/Query/UpdateQueryTest.php:474]: Aren't those meant to be assignment instead?

Comment (ADmad) [reply tests/TestCase/Database/Query/UpdateQueryTest.php:474]: Yes, it's used in the `skipIf()` call below.

File: tests/TestCase/View/CellTest.php [hunk 2]
```diff
@@ -110,7 +110,6 @@ public function testCellImplictRenderWithError(): void
 
         $cell = $this->View->cell('Articles::teaserList');
         $cell->viewBuilder()->setTemplate('nope');
-        (string)$cell;
```
Comment (ADmad) [inline tests/TestCase/View/CellTest.php:113]: The string casting is to implicitly trigger the `Cell::render()` call, it's needed.

**Comment by ADmad** [comment]
I have found dead code checks in most tools to be wonky.

**Comment by Harfusha** [comment]
Yeah this broke stuff, i will close the PR and add this rule to ignored ones. Will you fix the test in another PR?

**Pull request closed without merging**